standard/statements.md

```diff
@@ -2018,6 +2054,101 @@ using (ResourceType rN = eN)
 >
 > *end example*
 
+A `using` statement of the form
+
+```csharp
+await using (ResourceType resource = «expression» ) «statement»
+```
+
+corresponds to the expansions shown above with `IAsyncDisposable` instead of `IDisposable`, `DisposeAsync` instead of `Dispose`, and the `Task` returned from `DisposeAsync` is `await`ed. An `await using` of the form:
+
+```csharp
+await using (ResourceType resource = «expression» ) «statement»
+```
+
+corresponds to
+
+```csharp
+{
+    ResourceType resource = «expression»;
+    try
+    {
+        «statement»;
+    }
+    finally
+    {
+        IAsyncDisposable d = (IAsyncDisposable)resource;
+        if (d != null)
+        {
+            await d.DisposeAsync();
+        }
+    }
+}
+```
+
+When `ResourceType` is a reference type that implements `IAsyncDisposable`. Other expansions for `await using` perform similar substitutions from the synchronous `Dispose` method to the asynchronous `DisposeAsync` method.
```

Comment:
If this is meant to be a condition of the expansion, I think it would definitely be better *before* the expansion. (At the moment it's written as if it's a standalone sentence, but it doesn't really work like that.)